numpy/polynomial/polyutils.py

```diff
@@ -543,13 +543,18 @@ def _valnd(val_f, c, *args):
         See the ``<type>val<n>d`` functions for more detail
     """
     try:
-        args = tuple(np.array(args, copy=False))
-    except Exception:
+        # Automatic object array creation from ragged lists deprecated in 1.18,
+        # so pre-emptively raise. Change to correct exception when deprecation is removed
+        with warnings.catch_warnings():
+            warnings.filterwarnings('error', 'Creating an ndarray with automatic',
+                                    DeprecationWarning)
+            args = tuple(np.array(args, copy=False))
+    except DeprecationWarning:
         # preserve the old error message
```

Comment:
Is your claim that this was previously unreachable? Did I screw up when I added `_valnd`?

Reply:
What should the behaviour be here? E.g. what happens if `args` were previoulsy broadcastable?

Reply:
I think @charris drew attention to this code being bad when I moved it, but I persuaded him to merge anyway since it didn't change.

Reply:
I am wondering if using `args = np.broadcast_arrays(*args)` might be good. Albeit, maybe there is some speed issue against it. `broadcast_arrays` is very slow for a few polynomials maybe.

Reply:
Checking out v1.17.5 and putting `xxx = xxx` just after the `exept Exception` passes all tests, so apparently an exception is never triggered in tests.